include/RAJA/internal/Iterators.hpp

```diff
@@ -41,13 +43,14 @@ class numeric_iterator
 {
 public:
   using value_type = Type;
+  using s_value_type = strip_index_type_t<Type>;
```

Comment:
Why is this change needed?

Reply:
> Editing the RangeSegment Constructor to take StorageT arguments throws a narrowing warning when constructing the begin and end iterators as they use the difference_type to store their value, causing issues with unsigned types again. It would seem the Iterator class will need to be modified to allow for the use of the value_type instead of difference_type for storing its value. 

Value needed to be stripped a well, similar to before.

Reply:
I don't understand why enforcing a signed DiffT has required this change to the underlying iterator.

Reply:
Because the value the iterator currently holds needs to be the same as the StorageT, storing the value as the DiffT can cause narrowing issues if the StorageT given to iterator is unsigned. DiffT is still used to modify the iterators value, it just no longer stores it as that.

Reply:
I don't think that `make_signed` should narrow though, if the signed type cannot represent the maximum distance between two `StorageT` then we could have a problem.

Reply:
@mdavis36 and I chatted about this a bit this afternoon.  While you're quite right @davidbeckingsale, it's a problem essentially every stl container, and every pair of pointers, has.  The `intptr_t` and `ptrdiff_t` types are frequently the same size, just with one being signed, and taking the difference of two pointers farther apart than PTRDIFF_MAX is UB... yay c++...

Anyway, my take on it is we could put in some debugging help if we want to do our due-diligence and catch the issue happening, but it's something we kinda just have to deal with allowing if the user asks for it.